private/bufpkg/bufmodule/module_data.go

```diff
@@ -32,16 +32,23 @@ type ModuleData interface {
 	// ModuleKey contains the ModuleKey that was used to download this ModuleData.
 	//
 	// The Digest from this ModuleKey is used for tamper-proofing. It will be checked against
-	// the actual data downloaded before Bucket() or DeclaredDepModuleKeys() returns.
+	// the actual data downloaded before Bucket() or DepModuleKeys() returns.
 	ModuleKey() ModuleKey
 	// Bucket returns a Bucket of the Module's files.
 	//
 	// This bucket is not self-contained - it requires the files from dependencies to be so.
 	//
 	// This bucket will only contain module files - it will be filtered via NewModuleData.
 	Bucket() (storage.ReadBucket, error)
-	// DeclaredDepModuleKeys returns the declared dependencies for this specific Module.
-	DeclaredDepModuleKeys() ([]ModuleKey, error)
+	// DepModuleKeys returns the dependencies for this specific Module.
+	//
+	// The dependencies are the same as that would appear in the buf.lock file.
```

Comment:
```suggestion
	// The dependencies are the same as that would appear in the v1 buf.lock file.
```
(v2 buf.lock is for workspaces, so would contain more deps than this module needs)